component/remote/http/http.go

```diff
@@ -189,10 +198,15 @@ func (c *Component) pollError() error {
 	ctx, cancel := context.WithTimeout(context.Background(), c.args.PollTimeout)
 	defer cancel()
 
-	req, err := http.NewRequest(http.MethodGet, c.args.URL, nil)
+	req, err := http.NewRequest(c.args.Method, c.args.URL, nil)
```

Comment:
Hi, @tpaschalis 

I added a validation condition. The validation uses `http.NewRequest` as validator, since NewRequest validation the HTTP method and parses the URL. I added a test which validates this.

Please keep in mind, a there are not a enum of "valid" HTTP Methods. We may have only 9 in mind, but there standardized HTTP Extensions that are using other than the defined HTTP methods in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods). I'm not really a fan of adding a limitation here which is not necessary. 

https://github.com/golang/go/blob/master/src/net/http/request.go#L821-L836